app/utils/peer.js

```diff
@@ -93,21 +107,318 @@ function initFileDelivery(port, callback) {
   callback();
 }
 
+function loadActivityPattern(callback,activityPath) { // asynchronouslly loads the Activity Pattern
+  activityPath = (typeof activityPath !== 'undefined') ?  activityPath : 'ActivityPattern.txt';
+
+  fs.readFile(activityPath, function(err, f){
+    if(err){ return console.error('Failed To load the Activity Pattern file on path: '+ activityPath );   }
+    callback(f.toString().split(','));
+  });
+}
+
+function createActivityPatternFile(callback,deltaMinutes, activityDays, activityPath) { // asynchronouslly Creates the Activity Pattern File
+
+  // Defaults :
+  deltaMinutes = (typeof deltaMinutes !== 'undefined') ?  deltaMinutes : 10; // update activity every 10 min
+  activityDays = (typeof activityDays !== 'undefined') ?  activityDays : 7; // activity for 1 week
+  activityPath = (typeof activityPath !== 'undefined') ?  activityPath : 'ActivityPattern.txt';
+
+  //calculated variables
+  const partsPerHour = Math.ceil(60/deltaMinutes);
+  const partsPerDay  = Math.ceil(24*60/deltaMinutes);
+  const activityParts = Math.ceil(activityDays * partsPerDay)
+
+  //  initalize your activity with all offline, Zero !
+  activity = Array.apply(null, Array(activityParts) ).map(Number.prototype.valueOf,0);
+
+  bufferToFile(activityPath,activity,()=>{ // Created Activity Pattern File
+    console.log('Successfuly Created Activity Pattern File !')
+    callback(activity);
+  });
+
+}
+
 function initHost( port, networkID, seed, callback) {
   initDHT( ip.address(), port, networkID, seed, () => {
     initFileDelivery(port, () => {
+
+      console.log('Tracking using the Defaults activityPath');
+
+      node.plugin(quasar);
+
+      node.quasarSubscribe('icanhost', (broadcast) => {
+          console.log('recieved sth');
+          loadActivityPattern((myactivity)=>{
+
+              var mycontent = {
+                  ip: ip.address(),
+                  port: port,
+                  id: node.identity.toString('hex'),
+                  activity: myactivity
+              }
+
+              var uploaderip = broadcast['ip'];
+              var uploaderport = broadcast['port'];
+              var shredsize = broadcast['shred_size'];
+
+              if(remaining_capacity > shredsize)
+              {
+                  const socket = socketclient(`http://${uploaderip}:${uploaderport}`);
+
+                  socket.on('connect', function (socket) {
+                      console.log('Connected!');
+                  });
+
+                  socket.on('send.success', (yo) => {
+                      console.log(yo);
+                      socket.disconnect();
+                  });
+
+                  socket.emit('subscriber', mycontent);
+              }
+          });
+      });
+      console.log('Subscribed');
```

Comment:
create a new function for this, 'initPubSub' or something, so we can abstract it. Then call it from within 'initHost'